### PR TITLE
Document bag pocket magic numbers.

### PR DESCRIPTION
**Pull request by Meowsy**

### CODE REVIEW

File: src/item_menu.c [hunk 1]
```diff
@@ -526,9 +526,9 @@ void GoToBagMenu(u8 bagMenuType, u8 pocketId, void ( *postExitMenuMainCallback2)
             gUnknown_0203CE58.location = bagMenuType;
         if (postExitMenuMainCallback2)
             gUnknown_0203CE58.bagCallback = postExitMenuMainCallback2;
-        if (pocketId <= 4)
+        if (pocketId <= POCKETS_COUNT - 1)
```
Comment (huderlem) [inline src/item_menu.c]: `if (pocketId < POCKETS_COUNT)`

File: src/item_menu.c [hunk 2]
```diff
@@ -455,60 +455,60 @@ extern const u16 gUnknown_0860F074[];
...
 void CB2_ChooseBerry(void)
 {
-    GoToBagMenu(4, 3, CB2_ReturnToFieldContinueScript);
+    GoToBagMenu(4, BERRIES_POCKET, CB2_ReturnToFieldContinueScript);
 }
 
 void sub_81AABF0(void(*callback)(void))
 {
-    GoToBagMenu(5, 3, callback);
+    GoToBagMenu(5, BERRIES_POCKET, callback);
```
Comment (huderlem) [inline src/item_menu.c]: The first parameter for `GotToBagMenu()` is essentially an index into `gReturnToXStringsTable`, if you're feeling up to documenting that, too.

Comment (Meowsy) [reply src/item_menu.c]: Nice find!

File: src/item_menu.c (hunk 1 above)
Comment (Meowsy) [reply src/item_menu.c]: Hahaha, I was _hoping_ that would be equivalent, but I forgot to actually check.

File: src/item_menu.c (hunk 2 above)
Comment (Meowsy) [reply src/item_menu.c]: Documented.

**Pull request merged**